eldritch/test_events.py

```diff
@@ -822,6 +820,107 @@ def testCombatChoice(self):
+    while self.state.turn_phase != 'mythos':
+      self.state.next_turn()
+      self.resolve_until_done()
+    place = self.state.places[place_name]
+    place.gate = self.state.gates.popleft()
+    monster = next(iter(self.state.monsters))
+    monster.place = place
+    self.char.place = place
+    self.state.event_stack.append(CloseLocation(place_name, 1))
+    self.resolve_until_done()
+
+    self.assertEqual(place.closed_until, self.state.turn_number + 2)
+    self.assertEqual(self.char.place, place)
+    self.assertEqual(monster.place, place)
+    self.assertFalse(place.closed)
+
+    self.state.event_stack.append(GateCloseAttempt(self.char, place_name))
+    with mock.patch.object(events.random, "randint", new=mock.MagicMock(return_value=5)):
+      fight_lore = self.resolve_to_choice(MultipleChoice)
+      fight_lore.resolve(self.state, fight_lore.choices[0])
+      self.resolve_until_done()
+
+    self.assertEqual(place.closed_until, self.state.turn_number + 2)
+    self.assertEqual(self.char.place.name, "Uptown")
+    self.assertEqual(monster.place.name, "Uptown")
+
+    current_turn = self.state.turn_number
+    while self.state.turn_number == current_turn:
+      self.state.next_turn()
+      self.resolve_until_done()
+
+    self.assertEqual(place.closed_until, self.state.turn_number + 1)
+    current_turn = self.state.turn_number
+    while self.state.turn_number == current_turn:
+      self.state.next_turn()
+    self.assertFalse(place.closed)
+
+  def testMoveToClosedLocation(self):
+    place_name = "Woods"
+    self.state.event_stack.append(CloseLocation(place_name))
+    self.char.place = self.state.places["Uptown"]
+
+    while self.state.turn_phase != 'movement':
+      self.state.next_turn()
+      self.resolve_until_done()
+
+    self.assertEqual(self.char.place.name, "Uptown")
+    self.assertEqual(self.char.movement_points, 4)
+    self.state.event_stack.append(MoveOne(self.char, self.state.places[place_name]))
+    self.resolve_until_done()
+    self.assertEqual(self.char.place.name, "Uptown")
+    self.assertEqual(self.char.movement_points, 4)
+
+    turn_number = self.state.turn_number
+    while self.state.turn_number < turn_number + 5:
+      self.state.next_turn()
+      self.resolve_until_done()
+    self.assertTrue(self.state.places[place_name].closed)
+
+  def testMoveFromClosedLocation(self):
+    place_name = "Merchant"
+    place = self.state.places[place_name]
+    while self.state.turn_phase != 'mythos':
+      self.state.next_turn()
+      self.resolve_until_done()
+    self.state.event_stack.append(CloseLocation(place_name, for_turns=1, evict=False))
+    self.char.place = place
+    self.resolve_until_done()
+
+    while self.state.turn_phase != 'movement':
+      self.state.next_turn()
+      self.resolve_until_done()
+    self.assertTrue(place.closed)
+    self.state.event_stack.append(
+      Sequence([MoveOne(self.char, self.state.places["Downtown"]), EndMovement(self.char)], self.char)
```

Comment:
Note: EndMovement is no longer a thing. You may have to resolve the CityMovement event (or just manually adjust the game turn and phase).